gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1534,8 +1558,14 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 				checkOrConvertIntegerKind(store, last, n.High)
 				checkOrConvertIntegerKind(store, last, n.Max)
 
-				// if n.X is untyped, convert to corresponding type
 				t := evalStaticTypeOf(store, last, n.X)
+				if t.Kind() == ArrayKind {
+					if n.X.addressability() == addressabilityStatusUnsatisfied {
```

Comment:
Do we need this check? if n.X is array literal, it cannot be sliced, so t.Kind == ArrayKind already ensure it's not addressable.

Reply:
Also consider this:

```go
package main

func main() {
	a := [1]int{1}
	println(a[:])
}
```
this should work but not.

Reply:
Yes, we need this check.  `t` is the type, not the expression.  So while `t` might be an array, `n.X` could be either a literal or a named expression or a function call, etc.  That is why, in the case of arrays, we need to check addressability before slicing because an array value without an address can't be sliced.

The example you gave should work.  Can you make sure you have the most up to date changes and run it again?